lib/puppet/provider/oneview_server_profile/c7000.rb

```diff
@@ -24,14 +24,17 @@
   mk_resource_methods
 
   def exists?
-    @data = data_parse
-    empty_data_check([:found, :get_available_targets, :get_available_networks, :get_available_servers, :get_compliance_preview,
-                      :get_messages, :get_profile_ports, :get_transformation, :absent])
+    super([:found, :get_available_targets, :get_available_networks, :get_available_servers,
+           :get_compliance_preview, :get_messages, :get_profile_ports, :get_transformation, :absent])
+
     # gets the connections' uris
     connections_parse if @data['connections']
     # gets the hash of filters for queries; in case it does not exist, query will be nil
     @query = @data.delete('query_parameters')
-    !@resourcetype.find_by(@client, @data).empty?
+
+    return @resourcetype.find_by(@client, @data).any? unless resource['ensure'] == :present
```

Comment:
For the idempotency it's needed to compare and update data, this is made by resource_update only when ensure is present.